### PR TITLE
Add a hierarchical facet (and show field) for the cho_type_facet field

### LINKED ISSUE #552: View for cho_has_type (hierarchical facet)
Depends on https://github.com/sul-dlss/dlme-transform/issues/155. DLME has three type values: `cho_edm_type`, `cho_has_type`, and `cho_type`. The second field is a subcategory of the first field (e.g. one value of `cho_edm_type` is Text which has possible `cho_has_type` values of 'manuscript', 'book', etc.). The view should reflect this hierarchy. `cho_type` is whatever value the raw data provides when we receive it from the data provider. This could be useful as well and should also be useable. These values are not standardized; they could be thought of as tags.

### PR DESCRIPTION
**Pull request by jkeck**
Closes #552 

## Why was this change made?
To add support for a type/sub-type hierarchical facet.

## Was the documentation (README, API, wiki, ...) updated?
n/a

facet
show-view

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -77,6 +77,7 @@ gem 'riiif', '~> 2.0'
 gem 'sitemap_generator'
 
 gem 'blacklight-gallery', '>= 0.3.0'
+gem 'blacklight-hierarchy', github: 'sul-dlss/blacklight-hierarchy'
```
Comment (jkeck) [inline Gemfile]: We need to cut a release that is BL 7 compatible.  Happy to do that as part of this.

**Comment by ggeisler** [comment]
@jkect 👍I was just writing this comment when you posted yours.

This is a great start but I was wondering about:

1. The alignment of the item counts. I think this has come up before but I forget what the issue is. Are there complications to getting the counts to be right-aligned so they line up with the non-hierarchical facet counts?

2. Maybe your screenshot is not current but we should be able to collapse/expand the "Sound" label to expose "Interview" and other potential child values, correct?

**Comment by jkeck** [comment]
@ggeisler 
For both 1 and 2 this is what I'm meaning about needing some ❤️. It looks like we've added some styling in SW to do things like float the counts and add the (FontAwesome) icons so we'll likely want to do the same here.

The facet items are togglable, just w/o the icons / other styling it's not obvious from the static screenshot.

**Comment by jkeck** [comment]
Added a little ❤️ 
salt


facet-twiddle

**Comment by mjgiarlo** [comment]
:fire: :fire: :fire:

**Pull request merged**